projects/pluto/src/main/java/nl/fairspace/pluto/web/IconsResource.java

```diff
@@ -0,0 +1,55 @@
+package nl.fairspace.pluto.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import nl.fairspace.pluto.config.dto.PlutoConfig;
+
+import static nl.fairspace.pluto.config.Urls.ICONS_PATH;
+
+@RestController
+@Slf4j
+@AllArgsConstructor
+public class IconsResource {
+
+    private final PlutoConfig plutoConfig;
+    /**
+     * GET  /api/iconsvg/{icon_name} : returns an icon of specified name if exists.
+     */
+    @GetMapping(value = ICONS_PATH + "{icon_name}", produces = "image/svg+xml")
+    public ResponseEntity<byte[]> iconsvg(@PathVariable String icon_name) {
+        try {
+            InputStream in = getSvgIconInputStream(icon_name);
```

Comment:
stream closing is missed, to avoid leaking add the stream initialization to try () as InputStreams implement AutoClosable

Reply:
true, I missed the closing part, thanks!